docs/RelayServer.md

```diff
@@ -0,0 +1,240 @@
+
+```
+createdb trustlinesdb
+```
+
+
+The rest of this tutorial assumes that the user running the relay and
+py-eth-index related commands has a working postgresql environment configured.
+All programs consider the PG* environment variables and will read ~/.pgpass for
+information about passwords. The [py-eth-index section](#py-eth-index) describes
+how to create the trustlines specific tables.
+
+### py-eth-index
+The [py-eth-index repository](https://github.com/trustlines-network/py-eth-index)
+contains a helper program that synchronizes the relevant information from the
+blockchain into a postgresql database.
+
+#### Installation of py-eth-index
+
+Clone the git repository:
+```
+cd ~
+git clone https://github.com/trustlines-network/py-eth-index
+```
+We need a virtualenv for this repository, so let's create it:
+```
+python3 -m venv ~/opt/py-eth-index; ~/opt/py-eth-index/bin/pip install -U pip
+```
+and install py-eth-index
+```
+cd ~/py-eth-index
+~/opt/py-eth-index/bin/pip install -c constraints.txt -r requirements.txt
+~/opt/py-eth-index/bin/pip install -c constraints.txt .
+```
+#### Initializing the database
+After the database has been created, the database must be initialized. This can
+be done with the following command:
+
+```
+~/opt/py-eth-index/bin/ethindex createtables
+```
+#### Importing the ABIs
+We need to import the ABIs from the trustline-contracts. trustlines-contracts is
+installed as a dependency of the relay server. Please run the following only
+after you've installed the relay server.
+
+```
+cp ~/opt/relay/trustlines-contracts/build/contracts.json ~
+~/opt/py-eth-index/bin/ethindex importabi
+```
+
+#### Importing events
+The following command will start importing all relevant events into the postgres
+database:
+
+```
+~/opt/py-eth-index/bin/ethindex runsync
+```
+This program will run forever.
+
+
+### relay server
+#### Prerequisites for the installation
+##### secp256k1
+One of the dependencies used in the relay server is the secp256k1 library. If
+you're using python 3.5 on linux you can skip the folowing step, since pypi
+contains binary packages for secp256k1. If not, you'll have to run the following
+to install the secp256k1 C library:
+
+```
+git clone https://github.com/bitcoin-core/secp256k1.git
+cd secp256k1
+./autogen.sh
+./configure --enable-module-recovery
+make
+sudo make install
+sudo ldconfig
+```
+
+##### solidity
+```
+curl -L -o $HOME/bin/solc https://github.com/ethereum/solidity/releases/download/v0.4.21/solc-static-linux && chmod +x $HOME/bin/solc
+```
+
+#### Installation of the relay server
+
+Again we need to clone the git repository, create a virtualenv and install into
+that. Please note that we can't reuse the py-eth-index virtualenv, since both
+projects do have conflicting dependencies.
+```
+cd ~
+git clone https://github.com/trustlines-network/relay
+python3 -m venv ~/opt/relay; ~/opt/relay/bin/pip install -U pip
+cd ~/relay;
+~/opt/relay/bin/pip install -c constraints.txt populus
```

Comment:
Especially this I would say looks for someone not that familiar with venvs and pip as if this is some part of the relay code.

Reply:
Sorry, I'm not sure what you mean here. Can you please clarify?

Reply:
I think it is hard to see here that you call pip of your virtual env and not something of the relay server, as the executable lives in ~/opt/relay. I think we should name it better so that someone can see that it is a virtual env. So relay env or something like that.

Reply:
Can we just go on with absolute paths for now? (It's not like it's working). The next time I run through those instructions and will actually run them on a VM, I will change it to use activated virtualenvs.

Reply:
Just how I see it, if you feel it is better this way, we can leave it.